packages/vitest/src/runtime/run.ts

```diff
@@ -288,7 +287,17 @@ export async function startTests(paths: string[], config: ResolvedConfig) {
 
   await runFiles(files, config)
 
-  takeCoverage()
+  // TODO: How could this __VITEST_COVERAGE__ specific logic be passed from pool.ts to here
+  // so that it would only be declared inside coverage/istanbul.ts?
+  // Passing globalThis to onFilesRun does not work
+
+  // @ts-expect-error -- untyped global
+  const coverage = globalThis.__VITEST_COVERAGE__
+
+  // TODO: Not sure if this will work for c8. Previously v8.takeCoverage() was called
+  // here inside the worker. Now, this will simply inform the main process to call it.
+
+  rpc().onFilesRun(coverage)
```

Comment:
I think we can keep `takeCoverage` c8 specific logic here. If v8 coverage is not enabled `v8.takeCoverage` should be a noop

Reply:
Alright it seems we need to add the explicit `takeCoverage()` here. Otherwise it will cause random flakiness where the coverage report is partially missed, as reported by @wight554 in https://github.com/vitest-dev/vitest/pull/1676#issuecomment-1195793478.

> 4. c8 randomly ignores coverage for 1 file in 100% covered code when using c8 with these changes